x-pack/filebeat/input/salesforce/state.go

```diff
@@ -0,0 +1,45 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package salesforce
+
+import (
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// state is the state of the salesforce module. It is used to watermark the state
+// to avoid pulling duplicate data from Salesforce. The state is persisted separately
+// for EventLogFile and Object.
+type state struct {
+	Object       dateTimeCursor `json:"object,omitempty"`
+	EventLogFile dateTimeCursor `json:"event_log_file,omitempty"`
+}
+
+// dateTimeCursor maintains two distinct states for the event collection iteration.
+// The initial state represents the time of the first event, while the subsequent state denotes the time of the last event.
+// In certain SOQL queries for specific objects, sorting by all fields may not be feasible, and there may be no specific order.
+// This design allows users to exert maximum control over the iteration process.
```

Comment:
Let's link that piece of doc where it says that sorting on all objects is not available. I think for this we should ideally give an example. For a newcomer, it is not very helpful!

Reply:
Done